extensions/amp-video-docking/0.1/amp-video-docking.js

```diff
@@ -461,11 +455,23 @@ export class VideoDocking {
   register(video) {
     this.install_();
 
-    const {element} = video;
-    const fidelity = PositionObserverFidelity.HIGH;
-    this.getPositionObserver_().observe(element, fidelity, ({positionRect}) =>
-      this.updateOnPositionChange_(video, positionRect)
-    );
+    if (!this.intersectionObserver_) {
+      this.intersectionObserver_ = new IntersectionObserver(
+        (entries) => {
+          entries.forEach((entry) => {
```

Comment:
An element may technically have multiple entries. It may be worthwhile to ensure you don't call `updateOnPositionChange_` twice on the same element.

To do this best it helps to iterate on entries array backwards